packages/loader/container-loader/src/container.ts

```diff
@@ -833,10 +833,31 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this._resolvedUrl = resolvedUrl;
             await this.connectStorageService();
 
-            // upload blobs here (NYI)
-
-            // post summary here
             if (hasAttachmentBlobs) {
+                // upload blobs to storage
+                assert(!!this.loader.services.detachedBlobStorage, "assertion for type narrowing");
+                const redirectTable: Map<string, string | undefined>
+                    = new Map(this.loader.services.detachedBlobStorage.all().map((id) => [id, undefined]));
+                do {
+                    for (const id of this.loader.services.detachedBlobStorage.all()) {
+                        if (!redirectTable.has(id)) {
+                            redirectTable.set(id, undefined);
+                        }
+                    }
+                    for (const [localId, storageId] of redirectTable) {
+                        if (storageId === undefined) {
+                            const blob = await this.loader.services.detachedBlobStorage.readBlob(localId);
+                            const response = await this.storageService.createBlob(blob);
```

Comment:
If this fails, do we delete file? I think host would not have any file identification, so it can't delete this file. Something should have ability to clean it, we should not leave garbage.